apis/v1alpha1/shared_types.go

```diff
@@ -80,77 +96,111 @@ type Port struct {
-	// If Samelabels is Empty then nothing is selected.
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
+	// Pods defines a way to select a set of pods in
+	// a set of namespaces. Note that host-networked pods
+	// are not included in this type of peer.
+	//
+	// Support: Core
+	//
+	// +optional
+	Pods *NamespacedPod `json:"pods,omitempty"`
+	// Nodes defines a way to select a set of nodes in
+	// the cluster. This field follows standard label selector
+	// semantics; if present but empty, it selects all Nodes.
+	//
+	// Support: Extended
+	//
+	// <network-policy-api:experimental>
 	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	SameLabels []string `json:"sameLabels,omitempty"`
-
-	// NotSameLabels is used to select a set of Namespaces that do not have certain
-	// values for a set of label(s).
-	// To be selected a Namespace must have all of the labels defined in NotSameLabels,
-	// AND at least one of them must have different values than the subject of this policy.
-	// If NotSameLabels is empty then nothing is selected.
+	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
+	// Networks defines a way to select peers via CIDR blocks.
+	// This is intended for representing entities that live outside the cluster,
+	// which can't be selected by pods, namespaces and nodes peers, but note
+	// that cluster-internal traffic will be checked against the rule as
+	// well. So if you Allow or Deny traffic to `"0.0.0.0/0"`, that will allow
+	// or deny all IPv4 pod-to-pod traffic as well. If you don't want that,
+	// add a rule that Passes all pod traffic before the Networks rule.
+	//
+	// Each item in Networks should be provided in the CIDR format and should be
+	// IPv4 or IPv6, for example "10.0.0.0/8" or "fd00::/8".
+	//
+	// Networks can have upto 25 CIDRs specified.
+	//
+	// Support: Extended
+	//
+	// <network-policy-api:experimental>
 	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	NotSameLabels []string `json:"notSameLabels,omitempty"`
+	// +listType=set
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=25
+	Networks []CIDR `json:"networks,omitempty"`
 }
 
-// NamespacedPodPeer defines a flexible way to select Namespaces and pods in a
-// cluster. The `Namespaces` and `PodSelector` fields are required.
-type NamespacedPodPeer struct {
-	// Namespaces is used to select a set of Namespaces.
-	Namespaces NamespacedPeer `json:"namespaces"`
-
-	// PodSelector is a labelSelector used to select Pods, This field is NOT optional,
-	// follows standard label selector semantics and if present but empty, it selects
-	// all Pods.
-	PodSelector metav1.LabelSelector `json:"podSelector"`
-}
+// CIDR is an IP address range in CIDR notation (for example, "10.0.0.0/8" or "fd00::/8").
+// This string must be validated by implementations using net.ParseCIDR
+// TODO: Introduce CEL CIDR validation regex isCIDR() in Kube 1.31 when it is available.
```

Comment:
We can assume that now can't we?

Reply:
i think this is now updated with the latest kube rebase, this PR might need another pull